cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -454,6 +462,200 @@ struct PowerChecked {
   }
 };
 
+struct Sin {
+  template <typename T, typename Arg0>
+  static enable_if_integer<Arg0, T> Call(KernelContext*, Arg0 val, Status*) {
+    static_assert(std::is_same<T, double>::value, "");
+    return sin(val);
+  }
+  template <typename T, typename Arg0>
+  static enable_if_floating_point<Arg0, T> Call(KernelContext*, Arg0 val, Status*) {
+    static_assert(std::is_same<T, Arg0>::value, "");
+    return sin(val);
```

Comment:
Use *std::sin* and friends to be consistent and be clear that this is not an Arrow function (or is it?).

Reply:
Whoops, not sure why it even lets me call that without std::. Fixed.